storage/replica_test.go

```diff
@@ -1547,6 +1548,10 @@ func TestRangeResponseCacheStoredError(t *testing.T) {
 // can be committed/aborted before being heartbeat.
 func TestEndTransactionBeforeHeartbeat(t *testing.T) {
 	defer leaktest.AfterTest(t)
+	// Don't automatically GC the Txn record: We want to heartbeat the
+	// committed Transaction and compare it against our expectations.
+	// When it's removed, the heartbeat would recreate it.
```

Comment:
I'm slightly concerned about the prospect of heartbeats recreating transaction records that were committed and GC'd. This means that from the perspective of an outside observer (say, another node making multiple PushTxn calls), a transaction would appear to transition from the committed state to the pending state. I don't see any actual problems that this would cause, but it feels a little strange (and it could, for example, interact badly with a cache like the one proposed in #1041).